R/merge_sce_list.R

```diff
@@ -156,12 +166,71 @@ merge_sce_list <- function(sce_list = list(),
     metadata_list$sample_metadata <- sample_metadata
   }
 
-  # Create the merged SCE from the processed list ------------------
+
+  ## Handle altExps ------------------------------------------------------
+
+  # If we are including altExps, process them and save to list to add to merged SCE
+  merged_altexps <- list()
+  if (include_altexp) {
+
+    # First we need to determine the final column names of the merged_sce (not yet made)
+    #  for use in altExp code. Later we'll apply this order to the merged_sce itself.
+    merged_colnames <- sce_list |>
+      purrr::map(colnames) |>
+      unlist() |>
+      unname()
+
+    # Find all altExp names present in the SCE objects.
+    # We will prepare a merged altExp for each of these.
+    altexp_names <- sce_list |>
+      purrr::map_chr(
+        \(sce) altExpNames(sce)
+      ) |>
+      purrr::reduce(union)
+
+    for (altexp_name in altexp_names) {
+      # Determine which SCEs contain this altExp, and create list of those altExps
+      has_altexp_name <- sce_list |>
+        purrr::map_lgl(
+          \(sce, altexp_name) altexp_name %in% altExpNames(sce),
+          altexp_name
```

Comment:
```suggestion
        purrr::map_lgl(
          \(sce) altexp_name %in% altExpNames(sce))
```
Maybe I'm just missing something, but I think you want to just return a vector of logicals with whether or not the `altexp_name` is present? So you don't want to return the `altexp_name`, since what you had will return it regardless of if it's present or not.

Reply:
This definitely is returning a logical vector... `altexp_name` on line 196 is the second argument to the "function".
How about I add braces like `\(sce, altexp_name) {altexp_name %in% altExpNames(sce)},` for clarity?
```
> has_altexp_name <- sce_list |>
+     purrr::map_lgl(
+         \(sce, altexp_name) altexp_name %in% altExpNames(sce),
+         altexp_name
+     )
> has_altexp_name
 sce1  sce2  sce3 
FALSE  TRUE  TRUE 
```

Reply:
see https://github.com/AlexsLemonade/scpcaTools/pull/243#discussion_r1423055684!